### PR TITLE
refactor(npm): logging when overriding .npmrc

### PR DESCRIPTION
**Pull request by rarkins**
## Changes:

Refactors npm extract function to more explicitly log npmrc scenarios. 

## Context:

Technical debt refactoring

## Documentation (please check one with an [x])

- [ ] I have updated the documentation, or
- [x] No documentation update is required

## How I've tested my work (please tick one)

I have verified these changes via:

- [ ] Code inspection only, or
- [x] Newly added unit tests, or
- [ ] No new tests but ran on a real repository, or
- [ ] Both unit tests + ran on a real repository

### CODE REVIEW

File: lib/manager/npm/extract/index.spec.ts [hunk 1]
```diff
@@ -112,6 +112,34 @@ describe('manager/npm/extract', () => {
       );
       expect(res.npmrc).toBeDefined();
     });
+    it('ignores .npmrc when ignoreNpmrcFile', async () => {
+      fs.readLocalFile = jest.fn((fileName) => {
```
Comment (viceice) [inline lib/manager/npm/extract/index.spec.ts:116]: use mocked `fs` from `test/utils`

File: lib/manager/npm/extract/index.ts [hunk 2]
```diff
@@ -92,13 +92,20 @@ export async function extractPackageFile(
...
-    logger.debug('Using configured npmrc');
-  } else if (config.ignoreNpmrcFile) {
-    npmrc = '';
-  } else {
-    npmrc = await readLocalFile(npmrcFileName, 'utf8');
-    if (is.string(npmrc)) {
+  const npmrcFileContent = await readLocalFile(npmrcFileName, 'utf8');
+  if (is.string(npmrcFileContent)) {
+    if (config.ignoreNpmrcFile) {
+      logger.debug({ npmrcFileName }, 'Ignoring .npmrc file in repository');
```
Comment (viceice) [inline lib/manager/npm/extract/index.ts:98]: logging content?

File: lib/manager/npm/extract/index.ts [hunk 3]
```diff
@@ -116,9 +123,9 @@ export async function extractPackageFile(
           .filter((line) => !line.includes('=${'))
           .join('\n');
       }
-    } else {
-      npmrc = undefined;
     }
+  } else if (config.ignoreNpmrcFile) {
+    npmrc = '';
```
Comment (viceice) [inline lib/manager/npm/extract/index.ts:128]: Missing coverage

**Review by viceice — CHANGES_REQUESTED** [review]

File: lib/manager/npm/extract/index.ts (hunk 2 above)
Comment (rarkins) [reply lib/manager/npm/extract/index.ts:98]: Just file name

**Pull request closed without merging**